xds/internal/xdsclient/controller/controller.go

```diff
@@ -88,6 +88,21 @@ type Controller struct {
 	lrsClients map[string]*lrsClient
 }
 
+var interceptor func(target string, opts ...grpc.DialOption) (*grpc.ClientConn, error)
+
+// SetDialInterceptor sets an interceptor for the controller. The interceptor
+// can be used to manipulate the dial options or change the target if needed.
+// The SetDialInterceptor must be called before the controller is created
```

Comment:
Controller can be created anytime, not only at the beginning.
The interceptor set can still affect the controllers created later. So this is never a no-op.

Let's just say this should only be called by before gRPC (e.g. in an init()).

Reply:
Clarified the comment  that this should be called before gRPC, instead of the no-op comment.